button-merchant/src/main/java/com/usebutton/merchant/ButtonApiImpl.java

```diff
@@ -195,8 +195,9 @@ public Void postOrder(Order order, String applicationId, String sourceToken,
 
                 String email = customer.getEmail();
                 if (email != null) {
```

Comment:
You should handle the null check inside `ButtonUtil.sha256Encode()`; including converting to lowercase

Reply:
Single responsibility!!!